### PR TITLE
SLCORE-348 Synchronize plugins per connection

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/ConnectedSonarLintEngineImpl.java [hunk 1]
```diff
@@ -517,16 +489,16 @@ public ProjectStorageStatus getProjectStorageStatus(String projectKey) {
...
-        globalStores.deleteAll();
-      }
       if (globalAnalysisContainer != null && containerStarted) {
         globalAnalysisContainer.stopComponents(false);
         containerStarted = false;
       }
       if (pluginInstancesRepository != null) {
         pluginInstancesRepository.close();
       }
+      if (deleteStorage) {
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/ConnectedSonarLintEngineImpl.java:499]: :+1:

File: core/src/main/java/org/sonarsource/sonarlint/core/storage/StoredPlugin.java [hunk 2]
```diff
@@ -17,22 +17,41 @@
...
-public class PluginReference {
+import java.nio.file.Path;
+import org.sonarsource.sonarlint.core.serverapi.plugins.ServerPlugin;
+
+public class StoredPlugin {
+  private final String key;
+  private final String fileName;
   private final String hash;
-  private final String filename;
+  private final Path jarPath;
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/storage/StoredPlugin.java:29]: Do we need both fileName and jarPath?

File: its/tests/src/test/java/its/StandaloneTest.java [hunk 3]
```diff
@@ -68,7 +68,7 @@ public static void prepare() throws Exception {
     Map<String, String> globalProps = new HashMap<>();
     globalProps.put("sonar.global.label", "It works");
     StandaloneGlobalConfiguration config = StandaloneGlobalConfiguration.builder()
-      .addPlugin(new File("../plugins/global-extension-plugin/target/global-extension-plugin.jar").toURI().toURL())
+      .addPlugin(new File("../plugins/global-extension-plugin/target/global-extension-plugin.jar").toPath())
```
Comment (henryju) [inline its/tests/src/test/java/its/StandaloneTest.java:71]:
```suggestion
      .addPlugin(Paths.get("../plugins/global-extension-plugin/target/global-extension-plugin.jar")))
```

**Review by henryju — APPROVED** [review]

File: core/src/main/java/org/sonarsource/sonarlint/core/storage/StoredPlugin.java (hunk 2 above)
Comment (damien-urruty-sonarsource) [reply core/src/main/java/org/sonarsource/sonarlint/core/storage/StoredPlugin.java:29]: Nope :+1:

**Pull request merged**